src/hp/derp/derp.rs

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package derp implements the Designated Encrypted Relay for Packets (DERP)
+// protocol.
+//
+// DERP routes packets to clients using curve25519 keys as addresses.
+//
+// DERP is used by Tailscale nodes to proxy encrypted WireGuard
+// packets through the Tailscale cloud servers when a direct path
+// cannot be found or opened. DERP is a last resort. Both side
+// between very aggressive NATs, firewalls, no IPv6, etc? Well, DERP.
+
+use std::io::{Read, Write};
+use std::time::Duration;
+
+use anyhow::{bail, Result};
+use byteorder::{BigEndian, ReadBytesExt, WriteBytesExt};
+use bytes::BytesMut;
+
+// MaxPacketSize is the maximum size of a packet sent over DERP.
```

Comment:
please update the comments to remove the go doc requirement of repeating the name and convert them to /// doc comments